numba/tests/test_parallel_backend.py

```diff
@@ -243,6 +254,13 @@ class TestParallelBackendBase(TestCase):
         ]
         all_impls.extend(parfor_impls)
 
+    masks = [i for i in [1, 2, 4, 8, 16] if i <= config.NUMBA_NUM_THREADS]
```

Comment:
Is this too overboard? It increases the test time for this file from 4 minutes to 18 minutes on my 16 core machine.

Reply:
That's probably a bit too much :) I presume the cost is messing about orchestrating the tests opposed to the actual run time of the tests?

Reply:
That said, we usually limit the CI runner core count to 4 or 8 cores and then run multiple configs for big-core-count machines as it seems to give better throughput as there's swathes of serial tests which won't parallelise.

Reply:
How about just `masks = [1, 2]`?

Reply:
Sounds good as a start.